crypto.go

```diff
@@ -117,7 +117,13 @@ func generateCertificateVerify(handshakeBodies []byte, privateKey crypto.Private
 	switch p := privateKey.(type) {
 	case ed25519.PrivateKey:
 		// https://crypto.stackexchange.com/a/55483
-		return p.Sign(rand.Reader, hashed, crypto.Hash(0))
+		// [..] You do not have to hash the inputs to Ed25519 because hashing is already
+		// part of Ed25519 itself. If you do hash inputs in advance, you become vulnerable
+		// to collisions in the hash function you use. [..]
+		//
+		// ed25519.PrivateKey.Sign(): signs the given message with priv. Ed25519 performs
+		// two passes over messages to be signed and therefore cannot handle pre-hashed messages.
```

Comment:
Maybe better to refer https://pkg.go.dev/crypto/ed25519#PrivateKey.Sign
> Sign signs the given message with priv. Ed25519 performs two passes over messages to be signed and therefore **cannot handle pre-hashed messages**.

Reply:
I'm not sure I understand, that's already in the comment? Or do you mean remove the Stackexchange link entirely?

Reply:
I think it's better to refer only official API documents and RFCs rather than someone's stackexchange answer. (not very strong opinion)
At least we should add the source of reference of L124-125.